File: Day76/Part4/arena_partition.py
import sys

def solve():
    try:
        input_data = sys.stdin.read().split()
        if not input_data:
            return
        
        iterator = iter(input_data)
        try:
            L = int(next(iterator))
            t = int(next(iterator))
            k = int(next(iterator))
        except StopIteration:
            return
            
        # Logic: Can we place a wall before enemy crosses the arena?
        # Time to cross = L (assuming speed 1)
        # Time to get wall = t
        # Need wall count >= 1
        
        if k >= 1 and t < L:
            print("true")
        else:
            print("false")
            
    except Exception:
        pass

if __name__ == "__main__":
    solve()

File: Day76/Part4/arena_partition.cpp
#include <iostream>

using namespace std;

bool solve() {
    long long L, t, k;
    if (!(cin >> L >> t >> k)) return false;
    
    // Assuming Player at 0, Enemy at L (or vice versa).
    // Enemy moves at speed 1.
    // We need to place a wall before Enemy reaches Player.
    // Time to reach is L.
    // We can place wall at time t.
    // If t < L, we can place a wall at position X (where Enemy hasn't reached).
    // We need at least 1 wall.
    
    if (k >= 1 && t < L) {
        return true;
    }
    return false;
}

int main() {
    cout << (solve() ? "true" : "false") << endl;
    return 0;
}
